### PR TITLE
Security 1728

### PR DESCRIPTION
**Pull request by dtbaum**

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/vncrecorder/VncRecorderBuildWrapper.java [hunk 1]
```diff
@@ -364,7 +364,7 @@ public FormValidation doCheckVncServ(@AncestorInPath AbstractProject<?,?> projec
 			}
 
 			//				String vnsServCom = "Example for start of vncserver: " + value.split(":").length == 2 : ;
-			return FormValidation.okWithMarkup("<strong><font color=\"blue\">Please, make sure that your vncserer is running on '" + value  + "'</font></strong>");
+			return FormValidation.okWithMarkup("<strong><font color=\"blue\">Please, make sure that your vncserer is running on '" + Util.escape(value)  + "'</font></strong>");
```
Comment (daniel-beck) [inline src/main/java/org/jenkinsci/plugins/vncrecorder/VncRecorderBuildWrapper.java]: This is the only change I understand. What are the others for?

File: src/main/java/org/jenkinsci/plugins/vncrecorder/VncRecorderBuildWrapper.java [hunk 2]
```diff
@@ -350,7 +350,7 @@ public FormValidation doCheckVncServ(@AncestorInPath AbstractProject<?,?> projec
 			}
 			//	    		List<String> com;
 			DescriptorImpl DESCRIPTOR = Hudson.getInstance().getDescriptorByType(DescriptorImpl.class);
-			String vnc2swf = Util.nullify(DESCRIPTOR.getVnc2swf());
+			String vnc2swf = Util.escape(Util.nullify(DESCRIPTOR.getVnc2swf()));
```
Comment (daniel-beck) [inline src/main/java/org/jenkinsci/plugins/vncrecorder/VncRecorderBuildWrapper.java]: This one makes sense as well.

Comment (Wadeck) [reply src/main/java/org/jenkinsci/plugins/vncrecorder/VncRecorderBuildWrapper.java]: I even suggest to only do the escaping inside the errorWithMarkup as the escape is changing some of the characters you could potentially find in filePath, like space.

Using Util.xmlEscape is sufficient for "general" XSS protection like required here as it's only escaping the <> and &.

**Review by Wadeck — COMMENTED** [review]
In terms of security the PR is correcting the problem. More concerns about "bad example" for other developers as some of the changes are not necessary and could/will be copy-pasted by people without understanding what they are doing and so, spreading the confusion.

Also, the potential breaking behavior for file path with space could be annoying for some users.

File: src/main/java/org/jenkinsci/plugins/vncrecorder/VncRecorderBuildWrapper.java (hunk 2 above)
Comment (daniel-beck) [reply src/main/java/org/jenkinsci/plugins/vncrecorder/VncRecorderBuildWrapper.java]: Right, `Util#escape` is the weird one that converts newlines into `<br/>`, basically a "HTML as plain text" converter.

Comment (dtbaum) [reply src/main/java/org/jenkinsci/plugins/vncrecorder/VncRecorderBuildWrapper.java]: My intention was to secure all existing job parameters against XSS with Util.escape(), even if they are currently not yet output as HTML, as "general" XSS" protection. I agree that Util.escape() isn't a right way for it. As mentioned, I will take xmlEscape() instead and will update my PR.

**Comment by dtbaum** [comment]
@Wadeck , @daniel-beck: see new PR under https://github.com/jenkinsci/vncrecorder-plugin/pull/5. I am open for further suggestions: Do you think that this approach of general" XSS protection  of all job parameters is a bad idea?

**Comment by daniel-beck** [comment]
> Do you think that this approach of general" XSS protection of all job parameters is a bad idea?

It makes no sense, as XSS protection must be done only once, and only during output.

You'd be surprised if you ran `export FOO="<bar>"` in `bash` and `echo $FOO` produced `&lt;bar&gt;`, no? That's what you seem to be doing here.

**Comment by dtbaum** [comment]
This use case does not concern this plugin, but in general I agree. 
Here is the new PR with Util.xmlEscape() in output only:  https://github.com/jenkinsci/vncrecorder-plugin/pull/6

**Pull request closed without merging**